lib/commands/connect/info.js

```diff
@@ -23,18 +14,22 @@ module.exports = {
     ],
     needsApp: true,
     needsAuth: true,
-    run: cli.command(function (context, heroku) {
-      return api.withUserConnections(context.auth.password, context.app, context.flags, true, heroku).then(function(connections){
-          if (connections.length === 0) {
-            console.log("No connection found, requesting auth...");
-            api.requestAppAccess(context.auth.password, context.app).then(function() {
-              api.withUserConnections(context.auth.password, context.app, context.flags, false, heroku).then(function(connections){
-                connections_info(connections);
-              });          
-            });
-          } else {
-            connections_info(connections);
-          }
+    run: cli.command(co.wrap(function* (context, heroku) {
+      let connections = yield api.withUserConnections(context.auth.password, context.app, context.flags, true, heroku);
+
+      if (connections.length === 0) {
+        console.log("No connection found, requesting auth...");
+        yield api.requestAppAccess(context.auth.password, context.app);
+        let connections = yield api.withUserConnections(context.auth.password, context.app, context.flags, false, heroku);
```

Comment:
Ah, yes. Previously, I had kept the dual connections_info calls, so it didn't need to escape it. Didn't catch this when I tidied it up.